autoload/lsp/capabilities.vim

```diff
@@ -35,6 +35,10 @@ function! lsp#capabilities#has_workspace_symbol_provider(server_name) abort
     return s:has_bool_provider(a:server_name, 'workspaceSymbolProvider')
 endfunction
 
+function! lsp#capabilities#has_implementation_provider(server_name) abort
+    return s:has_bool_provider(a:server_name, 'implementationProvider')
```

Comment:
spec says `implementationProvider?: boolean | (TextDocumentRegistrationOptions & StaticRegistrationOptions);` but for now seems ok. we can support it in the future if there is an ask.